components/ws-daemon/pkg/content/initializer.go

```diff
@@ -390,26 +390,35 @@ func (rs *remoteContentStorage) Download(ctx context.Context, destination string
 
 	span.SetTag("URL", info.URL)
 
-	tempFile, err := os.CreateTemp("", "remote-content-")
+	// create a temporal file to download the content
+	tempFile, err := os.CreateTemp("", "remote-content-*")
 	if err != nil {
 		return true, xerrors.Errorf("cannot create temporal file: %w", err)
 	}
-	defer tempFile.Close()
-	defer os.Remove(tempFile.Name())
+	tempFile.Close()
 
 	args := []string{
 		"-x16", "-j12",
+		"--allow-overwrite=true", // rewrite temporal empty file
 		info.URL,
 		"-o", tempFile.Name(),
 	}
+
 	cmd := exec.Command("aria2c", args...)
-	var out []byte
-	out, err = cmd.CombinedOutput()
+	out, err := cmd.CombinedOutput()
 	if err != nil {
 		log.WithError(err).WithField("out", string(out)).Error("unexpected error downloading file")
 		return true, xerrors.Errorf("unexpected error downloading file")
 	}
 
+	tempFile, err = os.Open(tempFile.Name())
+	if err != nil {
+		return true, xerrors.Errorf("unexpected error downloading file")
```

Comment:
Nit: we should try to change error messages to be more distinct. For example, so it's different from L411 and L416.

```suggestion
		return true, xerrors.Errorf("unexpected error opening temp file")
```